packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -183,6 +184,7 @@ class ReactShallowRenderer {
 
     if (shouldUpdate) {
```

Comment:
Does ReactDOM call the callback if `shouldComponentUpdate` skips an update? I would expect so, but this PR doesn't. Can you verify this?

Maybe it's better to move the callback call [here](https://github.com/accordeiro/react/blob/11b931d3b5e386e2fe164d8c7b2494081beb661e/packages/react-test-renderer/src/ReactShallowRenderer.js#L105). Then you don't need to duplicate it in two branches.